internal/target/repository.go

```diff
@@ -537,28 +523,130 @@ func (r *Repository) UpdateTarget(ctx context.Context, target Target, version ui
 			"SessionMaxSeconds":      target.GetSessionMaxSeconds(),
 			"SessionConnectionLimit": target.GetSessionConnectionLimit(),
 			"WorkerFilter":           target.GetWorkerFilter(),
+			"Address":                target.GetAddress(),
 		},
 		fieldMaskPaths,
 		[]string{"SessionMaxSeconds", "SessionConnectionLimit"},
 	)
 	if len(dbMask) == 0 && len(nullFields) == 0 {
 		return nil, nil, nil, db.NoRowsAffected, errors.New(ctx, errors.EmptyFieldMask, op, "empty field mask")
 	}
-	var returnedTarget Target
+
+	// The Address field is not apart of the target schema in the database.
+	// It is apart of a different table called target_address, which is why
+	// the Address field must be filtered out of the dbMask & nullFields slices.
+	var updateAddress, deleteAddress bool
+	var filteredDbMask, filteredNullFields []string
+	for _, f := range dbMask {
+		switch {
+		case strings.EqualFold("Address", f):
+			updateAddress = true
+		default:
+			filteredDbMask = append(filteredDbMask, f)
+		}
+	}
+	for _, f := range nullFields {
+		switch {
+		case strings.EqualFold("Address", f):
+			deleteAddress = true
+		default:
+			filteredNullFields = append(filteredNullFields, f)
+		}
+	}
+
+	// If the Address field is the only present change, then we must still
+	// update the target's version because a target address is a child object
+	// of the target.
+	if (len(filteredDbMask) == 0 && len(filteredNullFields) == 0) && (updateAddress || deleteAddress) {
+		target.SetVersion(version + 1)
+		filteredDbMask = append(filteredDbMask, "Version")
+	}
+
+	oplogWrapper, err := r.kms.GetWrapper(ctx, projectId, kms.KeyPurposeOplog)
+	if err != nil {
+		return nil, nil, nil, db.NoRowsAffected, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get oplog wrapper"))
+	}
+
 	var rowsUpdated int
+	var returnedTarget Target
 	var hostSources []HostSource
 	var credSources []CredentialSource
-	_, err := r.writer.DoTx(
+	_, err = r.writer.DoTx(
 		ctx,
 		db.StdRetryCnt,
 		db.ExpBackoff{},
 		func(read db.Reader, w db.Writer) error {
-			var err error
+			var msgs []*oplog.Message
 			t := target.Clone()
-			returnedTarget, hostSources, credSources, rowsUpdated, err = r.update(ctx, t, version, dbMask, nullFields)
+			ticket, err := w.GetTicket(ctx, t)
+			if err != nil {
+				return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get ticket"))
+			}
+
+			var tOplogMsg oplog.Message
+			rowsUpdated, err = w.Update(ctx, t, filteredDbMask, filteredNullFields, db.NewOplogMsg(&tOplogMsg), db.WithVersion(&version))
 			if err != nil {
 				return errors.Wrap(ctx, err, op)
 			}
+			if rowsUpdated > 1 {
+				return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("updated target version and %d rows updated", rowsUpdated))
```

Comment:
Here the transaction will rollback and nothing will actually be updated, so it's better to just return something like `more than 1 resource would have been updated`

Reply:
I like your suggested error message a lot more, I will update the code.